crates/hotshot/src/traits/networking/push_cdn_network.rs

```diff
@@ -155,13 +158,18 @@ pub struct PushCdnNetwork<TYPES: NodeType> {
 
 /// The enum for the topics we can subscribe to in the Push CDN
 #[repr(u8)]
+#[derive(IntoPrimitive, TryFromPrimitive, Clone, PartialEq, Eq)]
 pub enum Topic {
     /// The global topic
     Global = 0,
     /// The DA topic
     DA = 1,
 }
 
+/// Implement the `TopicTrait` for our `Topic` enum. We need this to filter
+/// topics that are not implemented at the application level.
+impl TopicTrait for Topic {}
```

Comment:
Does this actually implement anything?

Reply:
The trait is an alias that makes sure it implements `TryFrom<u8>`, `Into<u8>`, `Clone`, `PartialEq`, etc to enable conversions to and from a u8

Reply:
It also uses the default implementation for a function that prunes topics based on whether or not they are convertible to the application-level type